media/Config/exportConfig.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var importConfigPath = '';
+
+// Get file name from the path.
+const getFileName = function(oneToolList) {
+  let filePath = importConfigPath;
+  // The default name of '.cfg' file will be changed depending on
+  // the model file or imported '.cfg' file.
+  if (filePath === '') {
+    filePath = getFirstInputFilePath(oneToolList);
+  }
+  let fileName = filePath.split(/[\/\\]/).reverse()[0];
+  fileName = fileName.split('.');
+  if (fileName.length > 1) {
+    fileName.pop();
+  }
+  fileName = fileName.join('.');
+  return fileName;
+};
+
+// Get path of model file from input path.
+const getFirstInputFilePath = function(oneToolList) {
+  for (let i = 0; i < oneToolList.length; i++) {
+    if (oneToolList[i].use === true) {
+      for (let j = 0; j < oneToolList[i].options.length; j++) {
+        if (oneToolList[i].options[j].optionName === 'input_path') {
+          return oneToolList[i].options[j].optionValue;
+        }
+      }
+    }
+  }
+  return 'one-build-template';
```

Comment:
why do this return `one-build-template` ?

Reply:
Because this is the default name of `.cfg` file.
If there aren't any imported file or model file, then `one-build-template` will be the default name.

Reply:
@mhs4670go , I'm not sure using `one-build-template` as default name... do you have any idea?

Reply:
@seanshpark Umm, there's no right name. Since, before calling `getFirstInputFilePath`, a validation function that checks if tools have `input_path` is called, `one-build-template` is unlikely to be used.

`one-build.template`? Just empty? Any name looks okay.

Reply:
I don't know how `getFirstInputFilePath` is used but  can `getFirstInputFilePath` return undefined or null as error?

Reply:
Well if the tool which doesn't have `input_path` option (like `one-codegen`, `one-profile`) is used, then `one-build-template` will be the default name.

Reply:
FYI, returned value is used to autocomplete the name when the user push `export configuration` button. So, if some tools have `input_path` like `inception.pb`, then autocompleted name would be `inception.cfg`(or `inception.pb.cfg`?). But, no tools have `input_path`, default name would become needed.

Reply:
> then one-build-template will be the default name.

Assume from `getFirstInputFilePath` is this `one-build-template` be the input path name, that I assume to be file name for the model ?

Reply:
> Assume from `getFirstInputFilePath` is this `one-build-template` be the input path name, that I assume to be file name for the model ?

Yes right!
So then which name would be better? `one-build.template`? or `one-build-template`?

Reply:
> Yes right!

OK thanks for confirmation. 

> one-build.template? or one-build-template?

I think both are strange for the model names.
I don't know how the UI works but is it possible to cancel the operation for this case? that is, return null or undefined as I wrote before. The caller can handle some how.

If not, from my experience "(no-name)' or "(undefined)" was used, but not sure for this case.

Reply:
> return null or undefined as I wrote before.

But in my opinion, returning default name seems necessary as @mhs4670go said.
> FYI, returned value is used to autocomplete the name when the user push `export configuration` button. So, if some tools have `input_path` like `inception.pb`, then autocompleted name would be `inception.cfg`(or `inception.pb.cfg`?). But, any tools don't have `input_path`, default name would become needed.


There are 3 cases when the user clicks `Export Configuration` button.

**case 1.** When user imported `.cfg` file before
`auto completed default name`  : imported `.cfg` file

**case 2.** When user used the tool which has `input_path`
`auto completed default name`  : name of model file

**case 3.** When user used the tool which doesn't have `input_path`
`auto completed default name`  : 'one-build-template'

So due to the **case 3**, returning null or undefined or "(no-name)" doesn't seem good I think..?

Reply:
(1) what does `used the tool ` mean? how does the user act for this?
(2) As I don't know the UI, it would b better to share how the UI scenario is.

Reply:
> (1) what does `used the tool ` mean? how does the user act for this?

I mean the user select the tools like `one-optimize` , `one-quantize`...

 > (2) As I don't know the UI, it would b better to share how the UI scenario is.

I hope these `GIF`s could help your understanding :)

> **case 1.** When user imported `.cfg` file before
`auto completed default name` : imported `.cfg` file

![case-1](https://user-images.githubusercontent.com/58547845/135585339-50ccfc76-b3e5-45bf-83dc-b37dc50e6a23.gif)

> **case 2.** When user used the tool which has `input_path`
`auto completed default name` : name of model file

![case-2](https://user-images.githubusercontent.com/58547845/135585357-1d8bb0d9-75b8-42db-b11b-4a1c86919ce0.gif)

> **case 3.** When user used the tool which doesn't have `input_path`
`auto completed default name` : 'one-build-template'

![case-3](https://user-images.githubusercontent.com/58547845/135585375-8be8cfab-4a56-43af-bbce-a13de4e316ce.gif)

Reply:
Thanks for the video! This string is used as for unnamed `.cfg` file to save.
And I have more confidence `one-build-template` is not appropriate. Especially the `template`
For me, this is like when you create a new document, like in MS-Word, edit it, and then save the document.
Default name may be "Untitled, Unnamed or just Document"

Reply:
> Especially the `template` For me, this is like when you create a new document, like in MS-Word, edit it, and then save the document.

Yeah now it seems weird to me.

> Default name may be "Untitled, Unnamed or just Document"

Then I'll set default name as "Untitled"!